### PR TITLE
Use updated stack size configuration for HTTP multi-threaded demo

### PR DESCRIPTION
**Pull request by sukhmanm**
The stack size config used for task creation was initially taken from the `amazon-freertos` demo, and was causing runtime issues on Microchip. This PR reverts the change and uses `democonfigDEMO_STACKSIZE` instead, to account for platform differences as well as evolving differences in demo design. The change has been verified in CI.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c [hunk 1]
```diff
@@ -577,14 +572,14 @@ static void prvHTTPDemoTask( void * pvParameters )
             /* Open request and response tasks. */
             xDemoStatus = xTaskCreate( prvRequestTask,
                                        "RequestTask",
-                                       httpexampleTASK_STACK_SIZE,
+                                       democonfigDEMO_STACKSIZE,
```
Comment (leegeth) [inline FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c:575]: What was the runtime issue that got fixed with this change?

**Review by cobusve — APPROVED** [review]

File: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c (hunk 1 above)
Comment (leegeth) [reply FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c:575]: Verified on the hardware that the demos were failing due to an allocation failure and the reduction in stack size helped fix the issue.

**Review by leegeth — APPROVED** [review]

**Pull request merged**